test/shuffling_openpmd_test.py

```diff
@@ -0,0 +1,179 @@
+
+        old = io.Series("Be_shuffled1.out.h5", io.Access.read_only)
+        new = io.Series("Be_REshuffled1.out.h5", io.Access.read_only)
+        for it_index, old_it in old.iterations.items():
+            new_it = new.iterations[it_index]
+            for mesh_name, old_mesh in old_it.meshes.items():
+                new_mesh = new_it.meshes[mesh_name]
+                for component_name, old_component in old_mesh.items():
+                    new_component = new_mesh[component_name]
+                    old_array = old_component[()]
+                    new_array = new_component[()]
+                    old.flush()
+                    new.flush()
+                    assert np.isclose(np.sum(
+                        np.abs(old_array-new_array)), 0.0, atol=accuracy)
+
+    def test_training(self):
+        test_parameters = mala.Parameters()
+        test_parameters.openpmd_configuration = {
+            'adios2': {
+                'engine': {
+                    'parameters': {
+                        'StatsLevel': 1
+                    }
+                }
+            }
+        }
+        test_parameters.data.data_splitting_type = "by_snapshot"
+        test_parameters.data.input_rescaling_type = "feature-wise-standard"
+        test_parameters.data.output_rescaling_type = "normal"
+        test_parameters.network.layer_activations = ["ReLU"]
+        test_parameters.running.max_number_epochs = 50
+        test_parameters.running.mini_batch_size = 40
+        test_parameters.running.learning_rate = 0.00001
+        test_parameters.running.trainingtype = "Adam"
+        test_parameters.verbosity = 1
+        test_parameters.data.use_lazy_loading = True
+
+        # Train without shuffling.
+        data_handler = mala.DataHandler(test_parameters)
+        from mala.datahandling.data_repo import data_repo_path
+        original_data_path = os.path.join(data_repo_path, "Be2")
+        data_handler.add_snapshot("Be_snapshot0.in.h5", original_data_path,
+                                  "Be_snapshot0.out.h5", original_data_path, "tr", snapshot_type='openpmd')
+        data_handler.add_snapshot("Be_snapshot1.in.h5", original_data_path,
+                                  "Be_snapshot1.out.h5", original_data_path, "va", snapshot_type='openpmd')
+        data_handler.prepare_data()
+
+        test_parameters.network.layer_sizes = [data_handler.input_dimension,
+                                               100,
+                                               data_handler.output_dimension]
+        test_network = mala.Network(test_parameters)
+        test_trainer = mala.Trainer(test_parameters, test_network,
+                                    data_handler)
+        test_trainer.train_network()
+        old_loss = test_trainer.final_validation_loss
+
+        # Shuffle.
+        test_parameters.data.shuffling_seed = 1234
+        data_shuffler = mala.DataShuffler(test_parameters)
```

Comment:
I think that reusing the `test_parameters` in such a way might be dangerous, as the old `parameters.snapshot_directories_list` will be taken over. This issue already exists in `shuffling_test.py` based on numpy.

Reply:
Good point, I can change that. That should be an easy fix.